utils/CMSApi/remotePublish.ts

```
@@ -0,0 +1,46 @@
+import { httpClient, StatusCode } from "configs/api/httpClient/HttpClient";
+import { CMS_API_JWT, CMS_API_VERSION, JWT, STATUS_DRAFT, STATUS_PUBLISH } from "constants/general";
+import { IError } from "interfaces";
+
+const updateJwt = async () => {
+  const result:{data:{token: string}} = await httpClient.post(`${process.env.NEXT_PUBLIC_BASE_URL}${process.env.NEXT_PUBLIC_BASE_ROUTE}${CMS_API_JWT}`, {
+    username: `${process.env.NEXT_PUBLIC_CMS_USERNAME}`,
+    password: `${process.env.NEXT_PUBLIC_CMS_PASSWORD}`
+  });
+    
+  if (result.data.token){
+    localStorage.setItem(JWT, result.data.token);
+  }
+};
+
+const createPost = async (CMS_URL:string, body: object, isDraft:boolean | undefined) => {
+  const JWT_TOKEN = localStorage.getItem(JWT);
+  await httpClient.post(`${process.env.NEXT_PUBLIC_BASE_URL}${process.env.NEXT_PUBLIC_BASE_ROUTE}${CMS_API_VERSION}/${CMS_URL}`, {
+    ...body,
+    status: isDraft ? STATUS_DRAFT : STATUS_PUBLISH
+  }, {
+    headers: {
+      Authorization: `Bearer ${JWT_TOKEN}`
+    }
+  });
+};
+
+export const remotePublish =  async (CMS_URL: string, body:object, isDraft?:boolean) => {
+
+  const JWT_TOKEN = localStorage.getItem(JWT);
+
+  if (!JWT_TOKEN){
+    await updateJwt();
+  }
+
+  try {
+    await createPost(CMS_URL, body, isDraft);
+  } catch (error:IError | unknown){
+    if ((error as IError).status === StatusCode.Forbidden || (error as IError).status === StatusCode.Unauthorized){
```

we dont need to handle it here. here we can simply throw an error, it will be handle on httpCleint.ts error switch/case file, where we can check if its regarding the posts, and put some logic there